markdown/extensions/codehilite.py

```diff
@@ -249,6 +249,8 @@ def __init__(self, **kwargs):
                              'Default: True']
             }
 
+        if kwargs.get('linenums') == 'inline':
+            self.config['linenums'][0] = kwargs['linenums']
         super(CodeHiliteExtension, self).__init__(**kwargs)
```

Comment:
This should be completely unnecessary. The call to `super(CodeHiliteExtension, self).__init__(**kwargs)` should already pass on the value. In fact, I don't see any reason to make any changes at all. Did you check whether this works before making the change?

Reply:
https://github.com/Python-Markdown/markdown/blob/a4d4b61b5ce4a7dd96ddc13c66b8716d3ad8eb51/markdown/extensions/__init__.py#L46

I checked. It would raise ValueError if `linenums` is a string.

Reply:
Ah right. So then you should just need to change the type to accept strings.

Reply:
https://github.com/Python-Markdown/markdown/blob/a4d4b61b5ce4a7dd96ddc13c66b8716d3ad8eb51/markdown/util.py#L100-L116

Only changing type to string would cause compatibility issue. Maybe someone wrote `'yes'` in his code and expected `linenums` to be `True`.

Reply:
So I forgot how types were defined. But it's [documented][1]:

> Sets a configuration setting for `key` with the given `value`. If `key` is unknown, a `KeyError` is raised. If the previous value of `key` was a Boolean value, then `value` is converted to a Boolean value. If the previous value of `key` is `None`, then value is converted to a Boolean value except when it is `None`. No conversion takes place when the previous value of `key` is a string.

So your change is a hack to change the existing value to a string and avoid calling `parseBoolValue`. I don't like the hack, but this is the problem with implied types.

[1] https://python-markdown.github.io/extensions/api/#configsettings

Reply:
Yes, it is a ugly way to avoid compatibility issue. The root of all this is `linenos` in pygments accept two types. Maybe there is a elegant way to fix it.

Reply:
What about add a new parameter `lineno_format` with two options `'table'` and `'inline'`?

Reply:
> What about add a new parameter `lineno_format` with two options `'table'` and `'inline'`?

That might be a reasonable solution. And the value would be ignored except when `linenums` is `True`.